### PR TITLE
Fixing android recovery in device_doctor.

### PR DESCRIPTION
**Pull request by yusuf-goog**
Fixing package name regex and adding prints.

Bug:flutter/flutter#135035

*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: cipd_packages/device_doctor/lib/src/android_device.dart [hunk 1]
```diff
@@ -394,7 +394,7 @@ class AndroidDevice implements Device {
 
     // Listen to the stdout and stderr streams
     LineSplitter().convert(result.stdout).forEach((data) {
-      final packageMatch = RegExp(r'package:(.+)$').firstMatch(data);
+      final packageMatch = RegExp(r'package:.*=(.+)$').firstMatch(data);
```
[the remark this answers is not in the document]
Comment (yusuf-goog) [reply cipd_packages/device_doctor/lib/src/android_device.dart:397]: This is for cache clear. The next block runs deletes for 3p packages only.

File: cipd_packages/device_doctor/lib/src/android_device.dart [hunk 2]
```diff
@@ -420,7 +424,7 @@ class AndroidDevice implements Device {
...
```
Comment (keyonghan) [inline cipd_packages/device_doctor/lib/src/android_device.dart:427]: Based on go/flutter-luci-playbook#android-install-failed-insufficient-storage
We want to uninstall only those flutter apps:
```
adb shell pm list packages -f | grep '/data/app/com.example\|/data/app/com.yourcompany\|flutter' | awk '{split($0,a,"="); print a[2]}' | xargs -I % adb shell pm uninstall %
```

Comment (yusuf-goog) [reply cipd_packages/device_doctor/lib/src/android_device.dart:427]: Here are what it found on one of our devices:

```
$ adb shell pm list packages -3
package:com.google.android.keep
package:com.google.android.apps.translate
package:io.flutter.examples.hello_world
package:com.google.android.apps.docs.editors.docs
package:com.google.android.apps.tachyon
package:com.google.android.apps.magazines
package:com.google.android.apps.walletnfcrel
package:com.google.android.apps.chromecast.app
package:com.google.android.apps.podcasts
package:com.google.android.apps.subscriptions.red
```

Comment (keyonghan) [reply cipd_packages/device_doctor/lib/src/android_device.dart:427]: As mentioned above by @godofredoc , this list seems contains some system packages and may cause crash. Can we try adding the grep to give it a shot?

**Review by keyonghan — APPROVED** [review]
LGTM

**Pull request merged**